src/manager/node.c

```diff
@@ -1082,6 +1091,13 @@ static int agent_request_callback(sd_bus_message *m, void *userdata, UNUSED sd_b
         return req->cb(req, m, ret_error);
 }
 
+int agent_request_cancel(AgentRequest *r) {
+        _cleanup_agent_request_ AgentRequest *req = r;
+        req->is_cancelled = true;
+
+        return req->cb(req, NULL, NULL);
```

Comment:
I am wondering if we can define a 
```C
typedef void (*agent_request_cancel_t)(void *userdata);
// ...
struct AgentRequest {
   // ...
   agent_request_cancel_t cancel;
```
similar to `agent_request_response_t` and pass it to `node_create_request`. Then we can do a 
```C
req->cancel(req->userdata); 
```

This would remove the cancel logic (which is just replying to the caller that the request got cancelled) from the callback logic received as a response from the agent. We'd need to provide two functions (for job_setup and plain sd bus messages), I think. 
What do you think?

Reply:
I actually considered this option, but somehow it appeared to me that less complicated callback functions come at a price of more complicated AgentRequest structure and more complicated `node_create_request()` call. I think these are two equally valid implementations. @alexlarsson what do you think is better?

Reply:
I think this is overkill for the current requirements, and its more likely that someone forgets to implement the cancel request or gets it wrong, and then its not tested much.

Reply:
Because the `node_create_request` would require the cancel to be specified, I think its more likely that the cancel check inside the callback gets forgotten (if there is one to be added). 
But I agree that its quite complicated to implement. So lets go with the current approach.